### PR TITLE
EZP-24512: Add Details tab in the location view

### PR DESCRIPTION
**Pull request by yannickroger**
Link: https://jira.ez.no/browse/EZP-24512

## Description
Adds a new tab in the location view to display details of the content.

## Screenshot
details

## Test
Manual and unit test

## TODO
- [x] Add the tab
- [x] Display information
- [x] Async loading of creator and last contributor
- [x] Display languages
- [x] Fix unit test
- [x] Add unit test
- [x] Figure out why the tap event in the test doesn't work (see //FIXME)
- [x] CSS
- [x] Screenshot, rebase and cleanup

### CODE REVIEW

File: Resources/config/yui.yml [hunk 1]
```diff
@@ -132,6 +132,7 @@ system:
                         - 'ez-templatebasedview'
                         - 'ez-actionbarview'
                         - 'ez-locationviewviewtabview'
+                        - 'ez-locationviewviewdetailstabview'
```
Comment (dpobel) [inline Resources/config/yui.yml]: to be consistent with `ez-locationviewviewtabview`, the detail tab view should be called `eZ.LocationViewDetailsTabView` and should be defined in the `ez-locationviewdetailstabview` module.

File: Resources/config/yui.yml [hunk 2]
```diff
@@ -149,6 +150,12 @@ system:
                 locationviewviewtabview-ez-template:
                     type: 'template'
                     path: %ez_platformui.public_dir%/templates/tabs/view.hbt
+                ez-locationviewviewdetailstabview:
+                    requires: ['ez-locationviewtabview', 'ez-rawcontentview', 'locationviewviewdetailstabview-ez-template']
```
Comment (dpobel) [inline Resources/config/yui.yml]: `ez-rawcontentview` should be removed from the dependencies.

File: Resources/public/js/views/ez-locationviewview.js [hunk 3]
```diff
@@ -332,6 +332,17 @@ YUI.add('ez-locationviewview', function (Y) {
...
                         }),
+                        //TODO need all these params ?
+                        new Y.eZ.LocationViewViewDetailsTabView({
+                            content: this.get('content'),
+                            location: this.get('location'),
+                            contentType: this.get('contentType'),
+                            config: this.get('config'),
+                            languageCode: this.get('languageCode'),
+                            priority: 1000,
+                            selected: true,
```
Comment (dpobel) [inline Resources/public/js/views/ez-locationviewview.js]: the details tab is not selected by default, so this should be false

File: Resources/public/js/views/ez-locationviewview.js [hunk 4]
```diff
@@ -332,6 +332,17 @@ YUI.add('ez-locationviewview', function (Y) {
...
                             bubbleTargets: this,
                         }),
+                        //TODO need all these params ?
+                        new Y.eZ.LocationViewViewDetailsTabView({
+                            content: this.get('content'),
+                            location: this.get('location'),
+                            contentType: this.get('contentType'),
+                            config: this.get('config'),
+                            languageCode: this.get('languageCode'),
+                            priority: 1000,
```
Comment (dpobel) [inline Resources/public/js/views/ez-locationviewview.js]: please change the priority

File: Resources/public/js/views/tabs/ez-locationviewviewdetailstabview.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+     * @constructor
+     * @extends eZ.LocationViewTabView
+     */
+    Y.eZ.LocationViewViewDetailsTabView = Y.Base.create('locationViewViewDetailsTabView', Y.eZ.LocationViewTabView, [], {
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template({
+                "currentVersion": this.get('content').get('currentVersion').get('versionNo'),
+                "locationId": this.get('location').get('locationId'),
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewviewdetailstabview.js]: to make this simpler, I would pass `this.get('content').toJSON()`, `this.get('location').toJSON()` and `this.get('content').get('currentVersion').toJSON()` to the template.

File: Resources/public/js/views/tabs/ez-locationviewviewdetailstabview.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        },
+
+    }, {
+        ATTRS: {
+            /**
+             * The title of the tab
+             *
+             * @attribute title
+             * @type {String}
+             * @default "View"
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewviewdetailstabview.js]: to be updated

File: Resources/public/js/views/tabs/ez-locationviewviewdetailstabview.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+                value: "Details",
+                readOnly: true,
+            },
+
+            /**
+             * The identifier of the tab
+             * 
+             * @attribute identifier
+             * @type {String}
+             * @default "view"
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewviewdetailstabview.js]: same here

File: Resources/public/templates/tabs/details.hbt [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+                {{#if location.invisible }}
+                    Invisible
+                {{ else }}
+                    Visible
+                {{/if}}
+            {{/if}}
+        {{/if}}
+    </p>
+    <p>Location remote ID: {{ location.remoteId }}</p>
+    <p>Object remote ID: {{ content.remoteId }}</p>
```
Comment (dpobel) [inline Resources/public/templates/tabs/details.hbt]: *Content remove ID* to be consistent with *Content ID* above

File: Resources/public/js/views/services/plugins/ez-userloadplugin.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    "use strict";
+    /**
+     * Provides the user load plugin
+     *
+     * @module ez-userloadplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * Object relation list load plugin. It sets an event handler to load contents
```
Comment (dpobel) [inline Resources/public/js/views/services/plugins/ez-userloadplugin.js]: to be updated

File: Resources/public/js/views/services/plugins/ez-userloadplugin.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+         * @protected
+         * @method _loadUser
+         * @param {Object} e loadUser event facade
+         */
+        _loadUser: function (e) {
+            var loadOptions = {api: this.get('host').get('capi')},
+                User = this.get('userModelConstructor'),
+                loadedUser = new User(),
+                attribute = e.attributeName
+                ;
```
Comment (dpobel) [inline Resources/public/js/views/services/plugins/ez-userloadplugin.js]: wrong placement

File: Resources/public/js/views/services/plugins/ez-userloadplugin.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                } else {
+                    e.target.set(attribute, loadedUser);
+                }
+            });
+        },
+    }, {
+        NS: 'userLoad',
+
+        ATTRS: {
+            /**
```
Comment (dpobel) [inline Resources/public/js/views/services/plugins/ez-userloadplugin.js]: nitpick: alignment of the * should be
```
/**
 *
```

File: Resources/public/js/views/services/plugins/ez-userloadplugin.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+            * User Model constructor
+            *
+            * @attribute userModelConstructor
+            * @type Y.eZ.User
```
Comment (dpobel) [inline Resources/public/js/views/services/plugins/ez-userloadplugin.js]: nope, a constructor function is a Function and the default value is indeed `Y.eZ.User`

File: Resources/public/js/views/tabs/ez-locationviewdetailstabview.js [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+     */
+    Y.eZ.LocationViewDetailsTabView = Y.Base.create('locationViewDetailsTabView', Y.eZ.LocationViewTabView, [Y.eZ.AsynchronousView], {
+        initializer: function () {
+            this._fireMethod = this._fireLoadUser;
+
+            this.after('creatorChange', function (e) {
+                this.render();
+            });
+
+            this.after('ownerChange', function (e) {
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewdetailstabview.js]: can be written:
```js
this.after(['creatorChange', 'ownerChange'], function () {
    this.render();
});
```

File: Resources/public/js/views/tabs/ez-locationviewdetailstabview.js [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+             * @event loadUser
+             * @param {String} userId Id of the author
+             * @param {String} attributeName Where to store the result
+             */
+            this.fire('loadUser', {
+                userId: creatorId,
+                attributeName: 'creator',
+            });
+
+            if (creatorId == ownerId) {
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewdetailstabview.js]: `===`

File: Resources/public/js/views/tabs/ez-locationviewdetailstabview.js [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+            },
+
+            /**
+             * The creator of the content
+             *
+             * @attribute creator
+             * @type {Object}
+             */
+            creator: {
+                valueFn: function () {
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewdetailstabview.js]: you should not create an instance of Y.eZ.User as the default value of those attributes as they will be overwritten by the plugin. (That means you have to handle a non object value in `render`)

File: Resources/public/templates/tabs/details.hbt [hunk 16]
```diff
@@ -0,0 +1,29 @@
+<div>
+    <p>Creator: {{contentCreator.name}} - {{content.publishedDate}}</p>
```
Comment (dpobel) [inline Resources/public/templates/tabs/details.hbt]: when the users are not yet loaded you should display a loader to the user as it's done in the other AsynchronousView usage in the application

File: Resources/public/templates/tabs/details.hbt [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+    <p>Last contributor: {{lastContributor.name}} - {{content.lastModificationDate}}</p>
+
+    <p>Version: {{ currentVersion.versionNo }}</p>
+    <p>
+        Translation: {{languageCount}}
+        {{#each translationList}}
+            - {{.}}
```
Comment (dpobel) [inline Resources/public/templates/tabs/details.hbt]: please structure the output as HTML so here you should generate a ul/li list.

File: Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+ */
+YUI.add('ez-userloadplugin-tests', function (Y) {
+    var tests, registerTest,
+        Assert = Y.Assert;
+
+    tests = new Y.Test.Case({
+        name: "eZ User Load Plugin event tests",
+
+        setUp: function () {
+            this.User = function () {
```
Comment (dpobel) [inline Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js]: you don't need to set it as a property of the test:
```
var = User = function () {};
```

File: Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+                this.set = tests._set;
+                this.load = tests._load;
+            };
+
+            this.userId = '/my/user/42';
```
Comment (dpobel) [inline Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js]: this does not need to be global so it's better to keep it in local variable in the test methods.

File: Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+            var isCorrectValue = false;
+            Assert.areSame(
+                'id',
+                name,
+                'method set should be called to set the id'
+            );
+            if ( value == tests.userId ){
+                isCorrectValue = true;
+            }
+            Assert.isTrue(isCorrectValue, 'The id should be the user id');
```
Comment (dpobel) [inline Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js]: why not using `Assert.areEqual` ?

File: Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+            this.id = value;
```
Comment (dpobel) [inline Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js]: not needed ?

File: Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+                    'The REST API client should be passed in the load options'
+                );
+                callback();
+            };
+
+            this.view.fire('loadUser',{
+                userId: this.userId,
+                attributeName: 'myUser'
+            });
+            Assert.areSame(
```
Comment (dpobel) [inline Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js]: this assertion is useless, the correctness of the userId is already checked in `_set`

File: Tests/js/views/services/plugins/ez-userloadplugin.html [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+            },
+            "ez-viewservicebaseplugin": {
+                requires: ['plugin', 'base'],
+                fullpath: "../../../../../Resources/public/js/views/services/plugins/ez-viewservicebaseplugin.js"
+            },
+            "ez-pluginregistry": {
+                requires: ['array-extras'],
+                fullpath: "../../../../../Resources/public/js/services/ez-pluginregistry.js"
+            },
+            "ez-contentmodel": {
```
Comment (dpobel) [inline Tests/js/views/services/plugins/ez-userloadplugin.html]: not needed ?

File: Tests/js/views/services/plugins/ez-userloadplugin.html [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+                requires: ['ez-restmodel', 'array-extras'],
+                fullpath: "../../../../../Resources/public/js/models/ez-contentmodel.js"
+            },
+            "ez-restmodel": {
```
Comment (dpobel) [inline Tests/js/views/services/plugins/ez-userloadplugin.html]: not needed as well

File: Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+    });
+
+    registerTest = new Y.Test.Case(Y.eZ.Test.PluginRegisterTest);
+    registerTest.Plugin = Y.eZ.Plugin.UserLoad;
+    registerTest.components = ['locationViewViewService'];
+
+    Y.Test.Runner.setName("eZ User Load Plugin tests");
+    Y.Test.Runner.add(tests);
+    Y.Test.Runner.add(registerTest);
+}, '', {requires: ['test', 'view', 'base', 'ez-userloadplugin', 'ez-contentmodel', 'ez-pluginregister-tests']});
```
Comment (dpobel) [inline Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js]: 'ez-contentmodel' not needed

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 26]
```diff
@@ -0,0 +1,287 @@
...
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-locationviewdetailstabview-tests', function (Y) {
+    var attributesTest,
+        renderTest,
+        changeEventTest,
+        fireLoadUserEventTest,
+        Assert = Y.Assert,
+        Mock = Y.Mock
+        ;
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: CS

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 27]
```diff
@@ -0,0 +1,287 @@
...
+
+            origTpl = this.view.template;
+            this.view.template = function () {
+                templateCalled = true;
+                return origTpl.apply(this, arguments);
+            };
+            this.view.render();
+            Y.Assert.isTrue(templateCalled, "The template should have been used to render this.view");
+        },
+    });
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: a test is missing on which the available variables in the template.

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 28]
```diff
@@ -0,0 +1,287 @@
...
+
+            Mock.expect(this.creatorMock, {
+                method: 'toJSON',
+                returns: {}
+            });
+
+            this.view = new Y.eZ.LocationViewDetailsTabView({
+                content: this.contentMock,
+                location: this.locationMock,
+                contentType: {},
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: there's no `contentType` attribute on that tab view.

File: Resources/public/js/views/ez-locationviewview.js [hunk 29]
```diff
@@ -334,6 +334,15 @@ YUI.add('ez-locationviewview', function (Y) {
                             selected: true,
                             bubbleTargets: this,
                         }),
+                        new Y.eZ.LocationViewDetailsTabView({
+                            content: this.get('content'),
+                            location: this.get('location'),
+                            config: this.get('config'),
+                            languageCode: this.get('languageCode'),
```
Comment (dpobel) [inline Resources/public/js/views/ez-locationviewview.js]: not needed

File: Resources/public/js/views/tabs/ez-locationviewdetailstabview.js [hunk 30]
```diff
@@ -0,0 +1,183 @@
...
+            },
+
+            /**
+             * The language code in which the content displayed
+             *
+             * @attribute languageCode
+             * @type {String}
+             * @writeOnce
+             */
+            languageCode: {
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewdetailstabview.js]: not needed

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 31]
```diff
@@ -0,0 +1,287 @@
...
+            };
+            this.view.render();
+            Y.Assert.isTrue(templateCalled, "The template should have been used to render this.view");
+        },
+    });
+
+    changeEventTest = new Y.Test.Case({
+        name: "eZ LocationViewDetailsTabView change event test",
+        setUp: function () {
+            this.view = new Y.eZ.LocationViewDetailsTabView({});
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: nitpick: if you don't pass any arg, you can omit the empty config object

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 32]
```diff
@@ -0,0 +1,287 @@
...
+                content: that.contentMock
+            });
+        },
+
+        tearDown: function () {
+            this.view.destroy();
+            delete this.view;
+        },
+
+        "Should fire loadUser Event with different owner and creator": function () {
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: actually you are testing that the `loadUser` event is fired twice so title should be named:
*Should fire the loadUser event twice with a different owner and creator*

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 33]
```diff
@@ -0,0 +1,287 @@
...
+            });
+
+            this.view.set('active', true);
+
+            Assert.isTrue(ownerCalled, "loadUser should have been called for the owner");
+            Assert.isTrue(creatorCalled, "loadUser should have been called for the creator");
+
+        },
+
+        "Should fire loadUser Event with same owner and creator": function () {
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: Should fire the loadUser event once if the owner and the creator are the same

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 34]
```diff
@@ -0,0 +1,287 @@
...
+
+                that.view.once('loadUser', function (e) {
+                    creatorCalled = true;
+                });
+            });
+
+            this.view.set('active', true);
+
+            Assert.isTrue(ownerCalled, "loadUser should have been called for the owner");
+            Assert.isFalse(creatorCalled, "loadUser should not be called for the creator");
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: Y.fail in the event handler instead

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 35]
```diff
@@ -0,0 +1,287 @@
...
+            });
+
+            Mock.expect(that.versionMock, {
+                method: 'get',
+                args: ['resources'],
+                returns: {Creator: this.creator}
+            });
+
+            this.view = new Y.eZ.LocationViewDetailsTabView({
+                content: that.contentMock
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: nitpick: this

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 36]
```diff
@@ -0,0 +1,287 @@
...
+                    creatorCalled = true;
+                });
+            });
+
+            this.view.set('active', true);
+
+            Assert.isTrue(ownerCalled, "loadUser should have been called for the owner");
+            Assert.isFalse(creatorCalled, "loadUser should not be called for the creator");
+
+            this.view.set('creator', 'Paul Mc Cartney');
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: you should test that in different function and put an object in the creator attribute.

**Comment by dpobel** [comment]
for the style, first I would structure the view result a bit differently to get a markup that looks like:

```html
<h2>Content details</h2>
<dl>
    <dt>Creator</dt>
    <dd>John Doe</dd>
    <dt>Last contributor</dt>
    <dd>John Doe</dd>
    <dt>Visibility</dt>
    <dd>Hidden</dd>
    <dt>Versions</dt>
    <dd>18</dd>
    <dt>Translations</dt>
    <dd>...</dd>
</dl>

<h2>Technical details</h2>
<dl>
    <dt>Content Id</dt>
    <dd>...</dd>
    <dt>Location Id</dt>
    <dd>...</dd>
    <dt>Content remote Id</dt>
    <dd>...</dd>
    <dt>Location remote Id</dt>
    <dd>...</dd>
</dl>
```

File: Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js (hunk 21 above)
Comment (yannickroger) [inline Tests/js/views/services/plugins/assets/ez-userloadplugin-tests.js]: it is called by the `_loadUser` function `loadedUser.set('id', e.userId);` 

And then we check that id has been set properly:
```
Assert.areSame(
    this.userId,
    this.view.get('myUser').id,
    "The view should get the user"
 );
```

File: Resources/public/js/views/tabs/ez-locationviewdetailstabview.js [hunk 37]
```diff
@@ -0,0 +1,170 @@
...
+                this.render();
+            });
+        },
+
+        render: function () {
+            var container = this.get('container'),
+                content = this.get('content'),
+                currentVersion = content.get('currentVersion'),
+                translationsList = currentVersion.getTranslationsList(),
+                creator=null,
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewdetailstabview.js]: nitpick CS: spaces around `=`

File: Resources/public/js/views/tabs/ez-locationviewdetailstabview.js [hunk 38]
```diff
@@ -0,0 +1,170 @@
...
+                owner=null;
+
+            if (this.get('creator')){
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewdetailstabview.js]: nitpick cs: space before `{`

File: Resources/public/templates/tabs/details.hbt [hunk 39]
```diff
@@ -0,0 +1,86 @@
...
+            <dd>{{contentCreator.name}} - {{content.publishedDate}}</dd>
+        {{/if}}
+    {{else}}
+        {{#if loadingError}}
+            <dd class="ez-asynchronousview-error ez-font-icon">
+                An error occurred while loading the creator.
+                <button class="ez-asynchronousview-retry ez-button ez-font-icon pure-button">Retry</button>
+            </dd>
+        {{else}}
+            <dd class="ez-font-icon ez-asynchronousview-loading">Loading the creator</dd>
```
Comment (dpobel) [inline Resources/public/templates/tabs/details.hbt]: you should not add the `ez-font-icon` and `ez-asynchronousview-loading` classes to the `dd` element, instead you should add a paragraph with such class, that way the `dd` element can be generated outside of the condition.

File: Resources/public/templates/tabs/details.hbt [hunk 40]
```diff
@@ -0,0 +1,86 @@
+<h2>Content details</h2>
+<dl class="ez-details-authors ez-asynchronousview">
+    <dt>Creator</dt>
+    {{#if contentCreator }}
+
+        {{#if loadingError}}
```
Comment (dpobel) [inline Resources/public/templates/tabs/details.hbt]: to simplify a bit the template, I would write something like:
```handlebars
{{#if loadingError}}
    <p class="ez-asynchronousview-error ez-font-icon">
        An error occurred while loading the image.
        <button class="ez-asynchronousview-retry ez-button ez-font-icon pure-button">Retry</button>
    </p>
{{else}}
    {{#if contentCreator }}
        {{contentCreator.name}} - {{content.publishedDate}}
    {{else}}
        <p class="ez-font-icon ez-asynchronousview-loading">Loading the creator</p>
    {{/if}}
{{/if}}
```

File: Resources/public/templates/tabs/details.hbt [hunk 41]
```diff
@@ -0,0 +1,86 @@
...
+            <dd class="ez-asynchronousview-error ez-font-icon">
+                An error occurred while loading the last contributor.
+                <button class="ez-asynchronousview-retry ez-button ez-font-icon pure-button">Retry</button>
+            </dd>
+        {{else}}
+            <dd class="ez-font-icon ez-asynchronousview-loading">Loading the last contributor</dd>
+        {{/if}}
+    {{/if}}
+</dl>
+<dl>
```
Comment (dpobel) [inline Resources/public/templates/tabs/details.hbt]: 2 dl elements ?

**Comment by yannickroger** [comment]
Ready for review @dpobel @mhyndle

**Comment by dpobel** [comment]
+1

**Comment by mhyndle** [comment]
+1

**Pull request merged**